File: cpp_10-99/88.cc
// C++
// leetcode 88
// https://leetcode-cn.com/problems/merge-sorted-array/

class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
    	int idx = m + n - 1, p1 = m - 1, p2 = n - 1;
    	while(idx >= 0) {
            if(p1 < 0) {
                nums1[idx--] = nums2[p2--];
            } else if(p2 < 0 || nums1[p1] > nums2[p2]) {
    			nums1[idx--] = nums1[p1--];
    		} else {
    			nums1[idx--] = nums2[p2--];
    		}
    	}
    	return;
    }
};

File: Go/Go_10-99/88.go
// Go
// leetcode 88
// https://leetcode.cn/problems/merge-sorted-array/

func merge(nums1 []int, m int, nums2 []int, n int)  {
	idx := m+n-1
	i, j := m-1, n-1
	for i >= 0 && j >= 0 {
		if nums1[i] >= nums2[j] {
			nums1[idx] = nums1[i]
			idx--
			i--
			continue
		}
		// nums1[i] < nums2[j]
		nums1[idx] = nums2[j]
		idx--
		j--
	}
	for j >= 0 {
		nums1[j] = nums2[j]
		j--
	}
	return
}